docs/uds-package-practices.md

```diff
@@ -82,6 +82,24 @@ To help maintain a UDS Package, it:
 
 - **Must** release its package to the `ghcr.io/defenseunicorns/packages/<group>` namespace as the application's name (i.e. `ghcr.io/defenseunicorns/packages/uds/mattermost`)
 
+## Versioning
+
+- **Must** be versioned as follows:
+```
+<upstream-app-version>-uds.<uds-sub-version>
+```
```

Comment:
Just noting that for applications we create ourselves this suffix wouldn't likely be necessary (UDS runtime would be a good example). We may want to qualify this with _if using an upstream_.

UDS Core also would violate this - and likely any other packages that pull in multiple applications (not sure if that is realistic in the future though). Those might be very edge case, but could be a reason to change to a `Should`?

Reply:
I think it also should only apply if the package has a canonical app version to track - IMO there are a few (rare) cases where we would want to package multiple "apps" together - for [uds-package-sigstore](https://github.com/defenseunicorns/uds-package-sigstore) there are many apps in one package because fulcio for example doesn't do much without ctlog and trillian as well (and those pieces stand alone don't mean much either).

Reply:
(I do think we could say that you **Must** follow this versioning scheme _or_ semantic versioning for your main release artifacts - also not sure but we could cover snapshots too)